server/src/services/itoService.ts

```diff
@@ -354,5 +386,43 @@ export default (router: ConnectRouter) => {
       console.log(`Successfully deleted all data for user: ${userId}`)
       return {}
     },
+
+    async getAdvancedSettings(_request, context: HandlerContext) {
+      const user = context.values.get(kUser)
+      const userId = user?.sub
+      if (!userId) {
+        throw new ConnectError('User not authenticated', Code.Unauthenticated)
+      }
+
+      const settings = await AdvancedSettingsRepository.findByUserId(userId)
+      if (!settings) {
+        // Return default settings if none exist
+        return create(AdvancedSettingsSchema, {
+          id: '',
+          userId: userId,
+          createdAt: new Date().toISOString(),
+          updatedAt: new Date().toISOString(),
+          llm: create(LlmSettingsSchema, {
+            asrModel: 'distil-whisper-large-v3-en',
+          }),
+        })
+      }
+
+      return dbToAdvancedSettingsPb(settings)
+    },
+
+    async updateAdvancedSettings(request, context: HandlerContext) {
+      const user = context.values.get(kUser)
+      const userId = user?.sub
+      if (!userId) {
+        throw new ConnectError('User not authenticated', Code.Unauthenticated)
+      }
```

Comment:
Also unrelated to this PR, but we should look into middleware possibilities for this userId check rather than doing it at the service method level.